src/libs/history.c

```diff
@@ -701,15 +701,11 @@ static void _lib_history_change_callback(gpointer instance, gpointer user_data)
 static void _lib_history_compress_clicked_callback(GtkWidget *widget, gpointer user_data)
 {
   const int32_t imgid = darktable.develop->image_storage.id;
-  if(!imgid) return;
+  if(imgid<0) return;
 
   dt_history_compress_on_image(imgid);
 
-  sqlite3_stmt *stmt;
-
-  // load new history and write it back to ensure that all history are properly numbered without a gap
-  dt_dev_reload_history_items(darktable.develop);
-  dt_dev_write_history(darktable.develop);
+/*  sqlite3_stmt *stmt;
```

Comment:
Either remove this code or add a clear comment about why it is kept otherwise this will be noise for others and anyway at some point it will get removed and nobody will remember why it is there :)

Reply:
Sure, i kept this as a reminder ... will be removed.